R2API/R2APIContentManager.cs

```diff
@@ -0,0 +1,595 @@
+            if (scp) {
+                AddSafeType(ref scp.entityStateTypes, new SerializableEntityStateType(type), scp.name);
+            }
+        }
+
+        private static void HandleGameObject(GameObject go, SerializableContentPack scp) {
+            try {
+                bool alreadyNetworked = false;
+                bool addedToAnyCatalogs = false;
+                if (go.GetComponent<CharacterBody>()) {
+                    AddSafe(ref scp.bodyPrefabs, go, scp.name);
+                    alreadyNetworked = true;
+                    addedToAnyCatalogs = true;
+                }
+                if (go.GetComponent<CharacterMaster>()) {
+                    AddSafe(ref scp.masterPrefabs, go, scp.name);
+                    alreadyNetworked = true;
+                    addedToAnyCatalogs = true;
+                }
+                if (go.GetComponent<ProjectileController>()) {
+                    AddSafe(ref scp.projectilePrefabs, go, scp.name);
+                    alreadyNetworked = true;
+                    addedToAnyCatalogs = true;
+                }
+                if (go.GetComponent<Run>()) {
+                    AddSafe(ref scp.gameModePrefabs, go, scp.name);
+                    alreadyNetworked = true;
+                    addedToAnyCatalogs = true;
+                }
+                //ror2 automatically networks prefabs that are in the arrays above this one. (since all of them already have network identities)
+                if (!alreadyNetworked && !PrefabAPI.IsPrefabHashed(go) && go.GetComponent<NetworkIdentity>()) {
+                    AddSafe(ref scp.networkedObjectPrefabs, go, scp.name);
+                    addedToAnyCatalogs = true;
+                }
+                //Modify this once dlc 1 comes out, as EffectDefs will be a game object array instead of an EffectDef array.
+                if (go.GetComponent<EffectComponent>()) {
+                    AddSafeType(ref scp.effectDefs, new EffectDef(go), scp.name);
+                    addedToAnyCatalogs = true;
+                }
+                if (!addedToAnyCatalogs) {
+                    throw new ArgumentException($"The GameObject {go.name} ({go.GetType()}) does not have any components that are supported by the ContentManager! \n" +
+                        $"If you think this is an Error and it should be supported, please file a bug report.");
+                }
+            }
+            catch (Exception e) { R2API.Logger.LogError(e); }
+        }
+
+        internal static void CreateContentPacks() {
+            if (!contentPacksCreated) {
+                R2API.Logger.LogInfo($"Generating a total of {BepInModNameToSerialziableContentPack.Values.Count} ContentPacks...");
+                List<ContentPack> contentPacks = new List<ContentPack>();
+                foreach (KeyValuePair<string, R2APIManagedContentPack> kvp in BepInModNameToSerialziableContentPack) {
```

Comment:
use kvp destructor `var (bepinModName, r2apiContentPack)` + typo in BepInModNameToSerialziableContentPack

Reply:
![image](https://user-images.githubusercontent.com/78170922/154097067-775f6c50-d247-46c8-8747-fdb065a6ee51.png)
![image](https://user-images.githubusercontent.com/78170922/154097093-2623b027-5c25-485a-92a8-91a5158d16f4.png)
I cant seem to properly use the kvp deconstructor in the foreach loop

Reply:
this should make it work, using missing ? https://github.com/xiaoxiao921/R2API/blob/master/R2API/MiscHelpers/KeyValuePairExtensions.cs#L16